yapapi/services.py

```diff
@@ -62,21 +61,24 @@ class ServiceState(statemachine.StateMachine):
     """State machine describing the state and lifecycle of a :class:`Service` instance."""
 
     # states
-    starting = statemachine.State("starting", initial=True)
+    pending = statemachine.State("pending", initial=True)
+    starting = statemachine.State("starting")
     running = statemachine.State("running")
     stopping = statemachine.State("stopping")
     terminated = statemachine.State("terminated")
     unresponsive = statemachine.State("unresponsive")
 
     # transitions
+    start = pending.to(starting)
     ready = starting.to(running)
     stop = running.to(stopping)
     terminate = terminated.from_(starting, running, stopping, terminated)
     mark_unresponsive = unresponsive.from_(starting, running, stopping, terminated)
+    restart = pending.from_(pending, starting, running, stopping, terminated, unresponsive)
```

Comment:
hmm... I wouldn't clear _all_ the transitions just now -> I believe that for now, we're only supporting a restart from `pending` and `starting` ...

Reply:
From the `ServiceState` POV it makes no difference.
The logic in `Cluster` indeed restarts only `pending`/`starting`, but is there any reason why `Cluster` logic should influence `ServiceState` logic?